### PR TITLE
Settings GUI: Move shadows dropdown to Shaders, remove Most Used

### PR DESCRIPTION
**Pull request by rubenwardy**
Part of #13476

The aim of the new settings GUI is to make it much easier for users to find and change settings based on a logical categorisation of settings. Most Used was only intended to be temporary, after this PR and #13704 it's no longer needed

The shadows dropdown is now on the Shaders page. The dropdown index is no longer persisted, it's instead detected based on the actual settings. Changing shadow settings manually shows "Custom" on the dropdown.

Note that I intend to hide the dynamic shadow settings (except this dropdown) as advanced, I've kept the settings as-is so that it's easier to test the behaviour of the dropdown - especially "Custom".

image

## To do

This PR is Ready for Review

## How to test

* Clear your minetest.conf
* Open settings GUI, go to Shaders
* Check the dynamic shadows check box, see that the drop says Medium
* Set dynamic shadows to a preset, see the values below change
* Change a value below, see the drop down set to "Custom"
* Set a preset, see the values change again. Check that custom is not available in the drop down

### CODE REVIEW

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -430,7 +430,7 @@ enable_dynamic_shadows (Dynamic shadows) bool false
 shadow_strength_gamma (Shadow strength gamma) float 1.0 0.1 10.0
 
 #    Maximum distance to render shadows.
-shadow_map_max_distance (Shadow map max distance in nodes to render shadows) float 200.0 10.0 1000.0
+shadow_map_max_distance (Shadow map max distance in nodes to render shadows) float 140.0 10.0 1000.0
```
Comment (lhofhansl) [inline builtin/settingtypes.txt:433]: Why the default change? That seems unrelated.

Comment (rubenwardy) [reply builtin/settingtypes.txt:433]: To match the medium preset

**Comment by lhofhansl** [comment]
Works for me.

(If now we can get rid of the "Open Settings" button somehow...)

**Review by SmallJoker — APPROVED** [review]
Changed a few settings back and forth. The "Shadows" options do work well. Whereas I think that the "Most used" tab would still be useful, it is hopefully soon superseded by the further rework tasks you already announced.

**Comment by rubenwardy** [comment]
Was hoping #13704 would get merged before this one. I suppose it doesn't matter as long as it's merged before release
